chain/test_chain.go

```diff
@@ -94,6 +95,10 @@ type TestChain struct {
 	// stateFactory used to construct state databases from db/root. Abstracts away the backing RPC when running in
 	// fork mode.
 	stateFactory state.MedusaStateFactory
+
+	ContractBytecodes map[string][]byte
```

Comment:
I would prefer having a pointer reference to the `CompiledContract`s over the bytecodes since it is more memory efficient and we can access the same information.